File: ProgrammingFundamentals/Java/02.ConditionalStatementsAndLoops/Lab/src/P03_BackIn30Min.java
import java.util.Scanner;
public class P03_BackIn30Min {
    public static void main(String[] args) {

        Scanner console = new Scanner(System.in);
        int hours = Integer.parseInt(console.nextLine());
        int minutes = Integer.parseInt(console.nextLine()) + 30;

        if(minutes >= 60) {
            hours++;
            minutes -= 60;
        }
        if(hours >= 24) {
            hours -= 24;
        }
        System.out.format("%d:%02d", hours, minutes);
    }
}

File: 01.ProgrammingFundamentals/C#/02.ConditionalStatementsAndLoops/Lab/P03_BackIn30Min/P03_BackIn30Min.cs
using System;

namespace P03_BackIn30Min
{
    class P03_BackIn30Min
    {
        static void Main(string[] args)
        {
            int hours = int.Parse(Console.ReadLine());
            int minutes = int.Parse(Console.ReadLine()) + 30;

            if (minutes >= 60)
            {
                hours++;
                minutes -= 60;
            }
            if(hours >= 24)
            {
                hours -= 24;
            }
            Console.WriteLine($"{hours}:{minutes:D2}");
        }
    }
}
